e2e/blog-test.sh

```diff
@@ -0,0 +1,165 @@
+
+    aws elbv2 register-targets --target-group-arn $TARGET_GROUP_ARN --targets Id=$(curl --silent http://169.254.169.254/latest/meta-data/instance-id),Port=$PORT
+
+    export LOAD_BALANCER_HOSTNAME=$(kubectl get service nlb-tls-app -ojson | jq -r ".status.loadBalancer.ingress[0].hostname")
+
+    LOAD_BALANCER_NAME=$(cut -d'.' -f1 <<<"$LOAD_BALANCER_HOSTNAME" | sed 's/\(.*\)-/\1\//')
+
+    LOAD_BALANCER_ARN=arn:aws:elasticloadbalancing:$AWS_REGION:$(aws sts get-caller-identity | jq -r ".Account"):loadbalancer/net/$LOAD_BALANCER_NAME
+
+    LISTENER_ARN=$(aws elbv2 describe-listeners --load-balancer-arn $LOAD_BALANCER_ARN | jq -r ".Listeners[0].ListenerArn")
+
+    aws elbv2 modify-listener --listener-arn $LISTENER_ARN --protocol TCP --port $PORT --default-actions Type=forward,TargetGroupArn=$TARGET_GROUP_ARN >/dev/null 2>&1
+
+}
+
+create_ca() {
+
+    export CA_ARN=$(aws acm-pca create-certificate-authority --certificate-authority-configuration file://$E2E_DIR/blog-test/ca_config.json --certificate-authority-type "ROOT" --query 'CertificateAuthorityArn' --output text)
+
+    aws acm-pca wait certificate-authority-csr-created --certificate-authority-arn $CA_ARN
+
+    aws acm-pca get-certificate-authority-csr --certificate-authority-arn $CA_ARN --output text --region us-east-1 >$E2E_DIR/blog-test/ca.csr
+
+    CERTIFICATE_ARN=$(aws acm-pca issue-certificate --certificate-authority-arn $CA_ARN --csr fileb://$E2E_DIR/blog-test/ca.csr --signing-algorithm SHA256WITHRSA --template-arn arn:aws:acm-pca:::template/RootCACertificate/V1 --validity Value=365,Type=DAYS --query 'CertificateArn' --output text)
+
+    aws acm-pca wait certificate-issued --certificate-authority-arn $CA_ARN --certificate-arn $CERTIFICATE_ARN
+
+    aws acm-pca get-certificate --certificate-authority-arn $CA_ARN --certificate-arn $CERTIFICATE_ARN --output text >$E2E_DIR/blog-test/cert.pem
+
+    aws acm-pca import-certificate-authority-certificate --certificate-authority-arn $CA_ARN --certificate fileb://$E2E_DIR/blog-test/cert.pem
+
+}
+
+delete_ca() {
+
+    aws acm-pca update-certificate-authority --certificate-authority-arn $CA_ARN --status "DISABLED"
+
+    aws acm-pca delete-certificate-authority --certificate-authority-arn $CA_ARN --permanent-deletion-time-in-days 7
+
+}
+
+clean_up() {
+    set +e
+
+    kubectl delete -f $E2E_DIR/blog-test/test-nlb-tls-app.yaml >/dev/null 2>&1
+
+    kubectl delete -f $E2E_DIR/blog-test/nlb-lab-tls.yaml >/dev/null 2>&1
+
+    kubectl delete -f $E2E_DIR/blog-test/test-cluster-issuer.yaml >/dev/null 2>&1
+
+    helm uninstall aws-load-balancer-controller -n kube-system >/dev/null 2>&1
+
+    delete_ca
+
+}
+
+install_aws_load_balancer() {
+    kubectl create serviceaccount aws-load-balancer-controller -n kube-system >/dev/null 2>&1
+    kubectl annotate serviceaccount aws-load-balancer-controller -n kube-system iam.amazonaws.com/role=$OIDC_IAM_ROLE >/dev/null 2>&1
```

Comment:
You can probably remove this since it seems like the creds are being passed in via env variables into the pod instead of using IRSA

Reply:
Removed serviceaccount